### PR TITLE
Refactor pkg/common, Fixes #11599

### LINKED ISSUE #11599: pkg/common - Refactor package
**Problem**
 
pkg/common - Refactor package
 
**Solution**
 
The `pkg/common` package holds only two functions, is badly named, and has no clear responsibility.
 
We need to refactor this package to either:
- Give it a proper name, README.md, and responsibility of its own
- Remove it and move its function somewhere more appropriate (to be defined)

### PR DESCRIPTION
**Pull request by runcom**
Fixes: #11599

Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

File: volumes/repository.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func (r *Repository) newVolume(path string, writable bool) (*Volume, error) {
 	var (
 		isBindMount bool
 		err         error
-		id          = common.GenerateRandomID()
+		id          = stringutils.GeneratestringutilsID()
```
Comment (cpuguy83) [inline volumes/repository.go]: Should be `stringutils.GenerateRandomID`

**Comment by cpuguy83** [comment]
Seems like you have a bunch of calls to `stringutils.GeneratestringutilsID` which should be `stringutils.GenerateRandomID`

File: volumes/repository.go (hunk 1 above)
Comment (runcom) [inline volumes/repository.go]: Already fixed locally :) waiting test to push! Thanks @cpuguy83

**Comment by runcom** [comment]
@cpuguy83 fixed all of them (vim and my fault)

**Comment by LK4D4** [comment]
Actually name for package is not giving any idea about functions in it. Only generaterandomstring is somehow related to `stringutils`.
Also I don't know why we don't use `GenerateRandomID` everywhere instead of `RandomString`.
@runcom Would you mind to rename your package to something like `stringid` and then we'll create issue about removing `RandomString`.

**Comment by runcom** [comment]
@LK4D4 It's ok, I'll do it.

I was just moving the other `makeRandomString` (from https://github.com/docker/docker/blob/master/integration-cli/utils.go#L304)  and `RandomString` (from https://github.com/docker/docker/blob/master/engine/env_test.go#L16) to this new `stringutils` so it will contain this other 2 new methods (`GenerateRandomAsciiString(n int)` and `GenerateRandomAlphaOnlyString(n int)`)

**Comment by LK4D4** [comment]
@runcom looks okay for me.

**Comment by runcom** [comment]
@LK4D4 What do you think about `pkg/id` with `GenerateRandomID` and `TruncateID` and `pkg/stringutils` with `GenerateRandomString()`, `GenerateRandomAsciiString(n int)` and `GenerateRandomAlphaOnlyString(n int)`?

**Comment by LK4D4** [comment]
@runcom I thought about `id` too :) But it is too common variable name, so it will be not very convenient to import. Overall idea is cool.

**Comment by runcom** [comment]
@LK4D4 let's call it `containerid` then, if I'm not totally wrong the id we are referring to is always about container id (or am I really wrong?)

**Comment by LK4D4** [comment]
@runcom Nope, it is image id as well. But if we will suppose such things, then pkg is not place for this. pkg should be packages not dependent on docker at all.

**Comment by runcom** [comment]
Alright, `stringid` is perfect then

**Comment by runcom** [comment]
@LK4D4 now there are two dinstinct pkgs, `stringutils` and `stringid`.

Theare are only two places where `stringutils` is needed to call `RandomString` (now `GenerateRandomString`) from the previous `common` pkg:
- https://github.com/docker/docker/blob/master/engine/engine.go#L81
- https://github.com/docker/docker/blob/master/utils/utils.go#L315

I'll open another issue and replace the calls to `GenerateRandomString` with `stringid.GenerateRandomID` as you suggested before to get rid of this function in `stringutils`.

The other functions in `stringutils` (`GenerateRandomAsciiString(n int)` and `GenerateRandomAlphaOnlyString(n int)`) are used around tests.

**Comment by LK4D4** [comment]
@runcom Cool

**Comment by LK4D4** [comment]
LGTM
ping @jfrazelle

**Comment by crosbymichael** [comment]
LGTM

thanks!

**Pull request merged**